src/v/archival/ntp_archiver_service.cc

```diff
@@ -1241,7 +1241,7 @@ static ss::sstring make_index_path(const remote_segment_path& segment_path) {
 ss::future<ntp_archiver_upload_result> ntp_archiver::upload_segment(
   model::term_id archiver_term,
   upload_candidate candidate,
-  std::vector<ss::rwlock::holder> segment_read_locks,
+  std::vector<ss::rwlock::holder>,
```

Comment:
I can't think this one should be changed to `/* segment_read_locks */`. I'm pretty sure that since this is a co-routine these locks in the parameter are actually serving as RAII locks, rather than "unused parameters" -- but can still squash the warning with the `/* */` approach. wdyt?

Reply:
How about making it super-obvious:
```c++
  std::vector<ss::rwlock::holder> segment_read_locks,
  std::optional<std::reference_wrapper<retry_chain_node>> source_rtc) {
    auto hold_locks{std::move(segment_read_locks});
```

Reply:
i like the obvious version. the purest in me thinks that people reading coroutine code should include the parameters in their mental model like they would an raii object such as hold_locks. either way is good with me!

Reply:
I feel like I had to check the lifetime of an unnamed parameter, just to be sure my mental model still applies. This is harder to question/ more obvious.

Reply:
sgtm